src/wallet/spend.cpp

```diff
@@ -819,6 +819,41 @@ static void DiscourageFeeSniping(CMutableTransaction& tx, FastRandomContext& rng
     }
 }
 
+static void ComputeChangeParams(CoinSelectionParams& coin_selection_params, CWallet& wallet, const CScript& scriptChange, CAmount recipients_sum, size_t recipients_size)
```

Comment:
in ea57562b4ab4071fd1a716f09e13d19ff68e99f1

I see how it's useful for the next commit, but the contract of the method is not really clear without reading the code.

Reply:
Guess that you are referring to the last two args?

I think that it isn't much of a problem because it is an internal static helper function.

They are needed for Knapsack min change target calculation. What maybe could do is to pass them as one: `mean_amount = std::floor(recipients_sum / recipients_size)` and document the function properly.

And also, in a follow-up, we could go further and move the else branch of the `scriptChange` block of code inside `ComputeChangeParams` too. That part is only needed when there isn't an existent change output.

Reply:
Let's document which part of `CoinSelectionParams` is modified and which is not.

Reply:
I mean `ComputeChangeParams` computes some of `CoinSelectionParams`, but it's not immediately obvious which ones without reading the code of the function

Reply:
> I mean `ComputeChangeParams` computes some of `CoinSelectionParams`, but it's not immediately obvious which ones without reading the code of the function

Ok. Added docs.
The function calculates all the `CoinSelectionParams` fields related to 'change'. No exceptions.